docs/dev/setup.md

```diff
@@ -0,0 +1,63 @@
+# Setup Guide
+
+This guide gives a simplified opinionated setup for developers contributing to rust-analyzer.
+
+## Prerequisites
+
+Since rust-analyzer is a Rust project, you will need to install Rust. You can download and install the latest stable version of Rust [here](https://www.rust-lang.org/tools/install).
+
+You will also need Visual Studio Code and Visual Studio Code Insiders for this workflow. You can download and install Visual Studio Code [here](https://code.visualstudio.com/Download) and you can download and install Visual Studio Code Insiders [here](https://code.visualstudio.com/insiders/).
+
+You will also need to install the rust-analyzer extension for Visual Studio Code and Visual Studio Code Insiders.
+
+More information about Rust on Visual Studio Code can be found [here](https://code.visualstudio.com/docs/languages/rust)
+
+## Step-by-Step Setup
+
+**Step 01**: Fork the rust-analyzer repository and clone the fork to your local machine.
+
+**Step 02**: Open the project in Visual Studio Code.
+
+**Step 03**: Open a terminal and run `cargo build` to build the project.
+
+**Step 04**: Install the language server locally by running the following command:
+
+```sh
+cargo xtask install --server --code-bin code-insiders --dev-rel
```

Comment:
```
cargo xtask install --server --dev-rel
```
no need to build the extension unless you use it

Reply:
This workflow requires building a local installation of the extension. This would allow the user to view `eprintln!` debug statements in the Rust Analyzer Language Server output.